### ISSUE TITLE
[bug] Images keep disappearing (but are still there, hidden)

### ISSUE THREAD

**Issue by Crazy-Unicooorn** [brandonp2412/Flexify#92]
One issue that I found some time ago now and is still present under version 1.1.75, is the images disappearing. They are still there, and to make one appear again, you have to simulate updating the image, and just hitting "save" does the trick. 

Video:
https://github.com/user-attachments/assets/0f366268-39c2-4390-8c5c-b753e652f7a2

I don't know how they disappear though… so I can't say how to replicate. However, it seems that this is an issue "per exercice", meaning that each exercice is affected independently. 

On…
Galaxy S20 FE 5G
Android 13
One UI 5.1

**Comment by brandonp2412** [comment]
Yep I need to make it more of a guarantee the images are always the same. Its stored against each individual set when it should probably be stored in a separate table with a join. A lot of the data is Desi normalized by default because I didn't have experience doing Drift joins but I've learned a lot since then.

**Comment by brandonp2412** [comment]
This happens when you create gym sets from the Plans page, I forgot to retrieve the image there

**Comment by Crazy-Unicooorn** [comment]
Hey @brandonp2412 , unfortuately this is still happening in version 1.1.78 😔

**Comment by brandonp2412** [comment]
Fuck it i'll fix it properly like i said i was going to in the beginning. This is what i get for being lazy

**Comment by Crazy-Unicooorn** [comment]
Lol, no pressure bro! It's not like it's a critical bug 😆 Take your time, the app is already pretty awesome and useful 😁

**Comment by brandonp2412** [comment]
Hey I know it's been a while but can anyone provide steps to re-produce this issue please? Last time I tried reproducing this again it didn't work. I'm not 100% sure how this happens just looking at the code

**Comment by brandonp2412** [comment]
So I managed to reproduce this but I'm not sure it should be fixed.
The images are saved per-exercise rather than being a single image. 
This lets people either use them as indicators of what the exercise is (the default is the previous one each time),
or they could use each image separately as a type of memoir. In the video what I assume happened was this:
1. You created an exercise with a set image
2. The next sets you entered didn't have one
3. Back on the graphs page it's showing the latest image (nothing)

Although this could be seen as a bug I like to think of it as a feature :)
